### PR TITLE
fix(rtk query): bad api export in fragment generated files [issue #7278]

### PR DESCRIPTION
**Pull request by hkasemir**
## Description

does not export an attempt to inject the rtk query api for graphql fragment files

Related # (issue)
https://github.com/dotansimha/graphql-code-generator/issues/7278

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots/Sandbox (if appropriate/relevant):
repro of issue: https://github.com/hkasemir/gql-codegen-issue

## How Has This Been Tested?

- [x] added a test for generating files for fragments in the rtk-query plugin test suite

**Test Environment**:
- OS: Mac OS Big Sur (11.2.3)
- `@graphql-codegen/typescript-rtk-query`: 2.2.1
- NodeJS: 17.2.0

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by hkasemir** [comment]
@charlypoly just tagging you here because I've seen you active here recently. I'm a new contributor, and would like review on this, but I can't actually request review from anyone via the github ui. Was thinking maybe someone would pick it up, but it's been over 3 weeks 😅 Would you be the right person to help me out or point me in the right direction?

**Comment by charlypoly** [comment]
Hi @hkasemir!

Thank you very much for your well-documented change proposal! (repro repo + code sandbox = ✨)

Those changes look good to me, however, I'll let @dotansimha take a final look before merging those changes.

Best,

**Review by charlypoly — APPROVED** [review]

**Review by n1ru4l — APPROVED** [review]

**Review by kamilkisiela — APPROVED** [review]

**Pull request merged**
